docs/api/js/utils.md

```diff
@@ -1,68 +1,785 @@
 ```typescript
-export const ZKSYNC_MAIN_ABI = new utils.Interface(require("../abi/IZkSync.json"));
+export const L2_ETH_TOKEN_ADDRESS = '0x000000000000000000000000000000000000800a';
 ```
 
-## IERC20 interface
+#### Bootloader
 
-Convenient when interacting with native tokens on zkSync.
+```ts
+export const BOOTLOADER_FORMAL_ADDRESS = '0x0000000000000000000000000000000000008001';
+```
 
-```typescript
-export const IERC20 = new utils.Interface(require("../abi/IERC20.json"));
+#### Contract deployer
+
+```ts
+export const CONTRACT_DEPLOYER_ADDRESS = '0x0000000000000000000000000000000000008006';
+```
+
+#### L1 messenger
+
+```ts
+export const L1_MESSENGER_ADDRESS = '0x0000000000000000000000000000000000008008';
 ```
 
-## Encoding paymaster params
+### Gas
 
-Utility method that returns the correctly formed `paymasterParams` object for the common [paymaster flows](../../dev/developer-guides/aa.md#built-in-paymaster-flows).
+#### `DEFAULT_GAS_PER_PUBDATA_LIMIT`
+
+- Use a large amount of gas for signing on layer 2.
+- The amount ensures any reasonable limit is accepted. 
+
+:::info
+- The operator is NOT required to use the actual value and can use any value up to that signed by the user.
+:::
 
 ```typescript
-export function getPaymasterParams(paymasterAddress: Address, paymasterInput: PaymasterInput): PaymasterParams
+export const DEFAULT_GAS_PER_PUBDATA_LIMIT = 50000;
 ```
 
-The definition of the `PaymasterInput` can be found [here](./types.md).
+#### `REQUIRED_L1_TO_L2_GAS_PER_PUBDATA_LIMIT`
 
-## Useful gas constants
-Currently, there is no method to accurately estimate the required `gasPerPubdataLimit`. That's why for now, it is highly recommended to provide the `DEFAULT_GAS_PER_PUBDATA_LIMIT`. Users are not charged more by providing it.
-Later on it will be possible to query the current recommended limit.
+Any amount is possible for L1->L2 transactions because the cost per gas is adjusted.
 
-```typescript
-const GAS_PER_PUBDATA_BYTE = 17;
+```ts
+export const REQUIRED_L1_TO_L2_GAS_PER_PUBDATA_LIMIT = 800;
+```
 
-// The large L2 gas per pubdata to sign. This gas is enough to ensure that
-// any reasonable limit will be accepted. Note, that the operator is NOT required to
-// use the honest value of gas per pubdata and it can use any value up to the one signed by the user.
-// In the future releases, we will provide a way to estimate the current gasPerPubdata.
-export const DEFAULT_GAS_PER_PUBDATA_LIMIT = 50000;
+## Functions
+
+### `applyL1ToL2Alias`
+
+Converts the address that submitted a transaction to the inbox on L1 to the `msg.sender` viewed on L2.
+
+#### Inputs
+
+- `address`: sender address as string.
+
+#### Outputs
+
+?? not sure how to describe what's coming out here
+
+```ts
+export function applyL1ToL2Alias(address: string): string {
+    return ethers.utils.hexlify(ethers.BigNumber.from(address).add(L1_TO_L2_ALIAS_OFFSET).mod(ADDRESS_MODULO));
+}
+```
+
+See also [`undol1tol2alias`](#undol1tol2alias).
+
+### `checkBaseCost`
```

Comment:
Unless there is a strict requiremet to document everything I would rather not document this method as it is rather low level and not needed for users

Reply:
My vote is leave it in, for a number of reasons. 
1. Even though users may never use it, they will see it mentioned here and there and perhaps wonder why it isn't documented, maybe go to git instead to look it up if they're interested. Our docs then look incomplete.
2. Our docs should be helpful to us too, new devs onboarding for example. Leaving info out again just makes them have to click around.
3. Completeness.

Reply:
@kailash-manasarovar but if we document every internal detail, then every change to the SDK (even its internal parts which should never be used by anyone) will be considered a breaking change. Then we should clearly document under each such internal method that it is internal and may be changed at any moment

Reply:
good point, removed, the user can look these ones up at their leisure